File: java/observer/CurrentConditionsDisplay.java
package observer;

public class CurrentConditionsDisplay implements ObserverInterface,DisplayInterface{

	private float temperature;
	private float humidity;
	private float airpressure;
	private SubjectInterface weatherData;
	
	public CurrentConditionsDisplay(SubjectInterface weatherData) {
		this.weatherData = weatherData;
		this.weatherData.registerObserver(this);
	}
	
	@Override
	public void update(float temp, float humidity, float airpressure) {
		this.temperature = temp;
		this.humidity = humidity;
		this.airpressure = airpressure;
		display();
	}

	@Override
	public void display() {
		System.out.println("Current Conditions: ");
		System.out.format("Temperature: %2.1f°C\n",temperature);
		System.out.format("Humidity: %2.0f%%\n",humidity);
		System.out.format("Airpressure: %2.0f Pa\n",airpressure);
	}
}

File: c++/observer_raw/CurrentConditionsDisplay.cpp
#include<iostream>
#include "CurrentConditionsDisplay.h"
#include "ObserverInterfaces.h"

CurrentConditionsDisplay::CurrentConditionsDisplay(ISubject* weatherData){
  this->weatherData = weatherData;
  weatherData->registerObserver(this);
}

void CurrentConditionsDisplay::display() const{
  std::cout << "\nCurrent Weather Conditions:" << std::endl;
  std::cout << "Temperature: " << this->temperature << " C " << std::endl;
  std::cout << "Humidity: " << this->humidity << " %" << std::endl;
  std::cout << "Pressure: " << this->pressure << std::endl;
}

void CurrentConditionsDisplay::update(float temperature, 
                                      float humidity, float pressure){
  this->temperature = temperature;
  this->humidity = humidity;
  this->pressure = pressure;
  this->display();
}
